janusgraph-core/src/main/java/org/janusgraph/graphdb/management/JanusGraphManager.java

```diff
@@ -270,5 +274,13 @@ private void updateTraversalSource(String graphName, Graph graph, GremlinExecuto
         graphManager.putTraversalSource(traversalName, traversalSource);
     }
 
+    private void removeBinding(String key) {
+        if (null != gremlinExecutor) {
+            final Bindings bindings = gremlinExecutor.getScriptEngineManager().getBindings();
+            if (bindings.get(key) != null) {
+                gremlinExecutor.getScriptEngineManager().getBindings().remove(key);
+            }
```

Comment:
`Bindings.get(key) == null`  is ambiguous. From the [docs](https://docs.oracle.com/javase/7/docs/api/javax/script/Bindings.html#get(java.lang.Object)):

> [...] if this map contains a mapping from a key k to a value v such that (key==null ? k==null : key.equals(k)), then this method returns v; otherwise it returns null. 

Therefore `!=null` means that the binding contains a key which is also not `null`. However, I did not research how a `null` entry in a `Binding` arises and which meaning it has in the context of the `GremlinExecutor` and therefore I'm not sure if such an entry should remain here or not. But from my naive understanding, @porunov is right here.